### PR TITLE
add accessibility fixes for the public facing side of Arches

### PR DESCRIPTION
**Pull request by apeters**
to test, make sure to do a yarn.install first

### CODE REVIEW

**Review by chiatt — CHANGES_REQUESTED** [review]
The notification icon is no longer green:
image

**Review by chiatt — CHANGES_REQUESTED** [review]
Layer visibility slider covers layer name
image

**Review by chiatt — CHANGES_REQUESTED** [review]
I'm unable to enter text into the resource-instance dropdown input. This prevents filtering needed to get to the 'Create New Resource' option at the bottom of the list.

Also, when trying to move through AFS workflow steps that use select2, errors occur preventing the workflow from advancing properly.

image

**Comment by chrabyrd** [comment]
The resource-instance dropdown does not show a dropdown arrow. This was tested with advanced-search.

Screenshot 2023-11-15 at 11 34 33 AM

However, I was able to enter text in the resource-instance-dropdown input.

**Comment by chrabyrd** [comment]
There is a double-border on resource instance dropdowns when no results are returned

Screenshot 2023-11-15 at 11 38 25 AM

File: arches/app/media/js/bindings/select2-query.js
Comment (chrabyrd) [inline arches/app/media/js/bindings/select2-query.js:80]: errant `console.log`

**Comment by chrabyrd** [comment]
> Also, when trying to move through AFS workflow steps that use select2, errors occur preventing the workflow from advancing properly.
> 
> image

I can verify this error, and it appears in the editor as well.

**Comment by chrabyrd** [comment]
The concept dropdown in RDM does not appear to be functional

Screenshot 2023-11-15 at 12 16 19 PM

**Comment by chrabyrd** [comment]
When a dropdown loads, it loses its placeholder text:

http://g.recordit.co/41hI7ORsDJ.gif

**Comment by chrabyrd** [comment]
manifest manager dropdown does not close when item is selected

http://g.recordit.co/wNnRrm5i6y.gif

**Comment by chrabyrd** [comment]
attempting to clear the entry in the manifest manager results in error:

http://g.recordit.co/vAwNARAdO1.gif

```
Uncaught TypeError: Cannot read properties of undefined (reading 'id')
```

**Comment by chrabyrd** [comment]
Manifest manager loads image service without hitting the button or saving the tile:

http://g.recordit.co/6V2uvNUF2X.gif

**Comment by chrabyrd** [comment]
cannot clear resource-instance-select in the resource editor:

http://g.recordit.co/79wdNW64zL.gif

**Comment by chrabyrd** [comment]
The dropdown in the multi-tile-step is no longer functional:

Screenshot 2023-11-15 at 12 44 55 PM

**Comment by apeters** [comment]
> The concept dropdown in RDM does not appear to be functional and has no placeholder text
> 
> Screenshot 2023-11-15 at 12 16 19 PM

fixed by > ae14a1f

**Comment by apeters** [comment]
> manifest manager dropdown does not close when item is selected
> 
> http://g.recordit.co/wNnRrm5i6y.gif

> attempting to clear the entry in the manifest manager results in error:
> 
> http://g.recordit.co/vAwNARAdO1.gif
> 
> ```
> Uncaught TypeError: Cannot read properties of undefined (reading 'id')
> ```


both fixed by > [2b12e66](https://github.com/archesproject/arches/pull/10265/commits/2b12e66be4d08e8d74d22eaf60871bad40f68e43)

**Comment by apeters** [comment]
> Manifest manager loads image service without hitting the button or saving the tile:
> 
> http://g.recordit.co/6V2uvNUF2X.gif

this operates the same way in dev/7.5.x

**Comment by apeters** [comment]
> The resource-instance dropdown does not show a dropdown arrow. This was tested with advanced-search.
> 
> Screenshot 2023-11-15 at 11 34 33 AM
> However, I was able to enter text in the resource-instance-dropdown input.

fixed by: [1cfab6b](https://github.com/archesproject/arches/pull/10265/commits/1cfab6b00238e090dbb664ec1a74ab36ad1e582c)

**Comment by apeters** [comment]
> There is a double-border on resource instance dropdowns when no results are returned
> 
> Screenshot 2023-11-15 at 11 38 25 AM

fixed by: [07f73ff](https://github.com/archesproject/arches/pull/10265/commits/07f73ffb7c0944dd785b27dbf9d4e1b2f97a2b23)

File: arches/app/media/js/bindings/select2-query.js
Comment (apeters) [reply arches/app/media/js/bindings/select2-query.js]: fixed by: [ad54fbc](https://github.com/archesproject/arches/pull/10265/commits/ad54fbc842ee7a69f0cbd9214479ec42078adbf9)

**Comment by apeters** [comment]
> When a dropdown loads, it loses its placeholder text:
> 
> http://g.recordit.co/41hI7ORsDJ.gif



> cannot clear resource-instance-select in the resource editor:
> 
> http://g.recordit.co/79wdNW64zL.gif

both fixed by: [880ef0f](https://github.com/archesproject/arches/pull/10265/commits/880ef0fdd82a59c1b89f676ca11f6c17681efa3f)

**Comment by apeters** [comment]
> The notification icon is no longer green: image

@chiatt this issue currently exists in dev/7.5.x

**Comment by apeters** [comment]
> The notification icon is no longer green: image

fixed by: [06fb7f4](https://github.com/archesproject/arches/pull/10265/commits/06fb7f4f36a444d02bb230508f26d9e6100e1f3a)

**Comment by apeters** [comment]
@chiatt this issue currently exists in dev/7.5.x



> Layer visibility slider covers layer name image

**Comment by chrabyrd** [comment]
The first step of the project collection workflow fails. This is a generic workflow step.

Screenshot 2023-11-17 at 3 12 55 PM

error:
```
Uncaught TypeError: Cannot read properties of undefined (reading 'resourceId')
    at eval (resource-instance-select.js:510:38)
```

config:
```
                            componentConfigs: [
                                {
                                    componentName: 'resource-instance-select-widget',
                                    uniqueInstanceName: 'select-project', /* unique to step */
                                    parameters: {
                                        graphids: [
                                            '0b9235d9-ca85-11e9-9fa2-a4d18cec433a'/* Project */
                                        ],
                                    },
                                },
                            ], 
```

**Comment by chrabyrd** [comment]
This is project-level, but the `Show Workflow Help` dropdown in the init-workflow plugin fails. Are there changes that need to be made to any select2Configs when migrating to select-woo?

**Comment by chrabyrd** [comment]
The fonts in the account section of the user profile manager are wrong. The values are small and the labels are big:

Screenshot 2023-11-17 at 3 22 29 PM

**Comment by chrabyrd** [comment]
nit: z-index conflict between dropdown menu and sidebar options

Screenshot 2023-11-17 at 3 29 44 PM

**Comment by chrabyrd** [comment]
Still noticing some of the behavior that when a dropdown loads it loses its placeholder text:

http://g.recordit.co/9oOhQddKiK.gif

**Comment by chrabyrd** [comment]
concept-select does not retain selection

http://g.recordit.co/ixpQG9Pxvj.gif

**Comment by apeters** [comment]
> I'm unable to enter text into the resource-instance dropdown input. This prevents filtering needed to get to the 'Create New Resource' option at the bottom of the list.
> 
> Also, when trying to move through AFS workflow steps that use select2, errors occur preventing the workflow from advancing properly.
> 
> image

fixed by https://github.com/archesproject/arches/pull/10265/commits/dc3827bb9ac503527a62ab01b282136754e489f1

**Comment by apeters** [comment]
> The first step of the project collection workflow fails. This is a generic workflow step.
> 
> Screenshot 2023-11-17 at 3 12 55 PM
> error:
> 
> ```
> Uncaught TypeError: Cannot read properties of undefined (reading 'resourceId')
>     at eval (resource-instance-select.js:510:38)
> ```
> 
> config:
> 
> ```
>                             componentConfigs: [
>                                 {
>                                     componentName: 'resource-instance-select-widget',
>                                     uniqueInstanceName: 'select-project', /* unique to step */
>                                     parameters: {
>                                         graphids: [
>                                             '0b9235d9-ca85-11e9-9fa2-a4d18cec433a'/* Project */
>                                         ],
>                                     },
>                                 },
>                             ], 
> ```

fixed in the latest

**Comment by apeters** [comment]
> Still noticing some of the behavior that when a dropdown loads it loses its placeholder text:
> 
> http://g.recordit.co/9oOhQddKiK.gif

I can't reproduce this

**Comment by chiatt** [comment]
> > Still noticing some of the behavior that when a dropdown loads it loses its placeholder text:
> > http://g.recordit.co/9oOhQddKiK.gif
> 
> I can't reproduce this

I see the same behavior, but only the first time the card is loaded. Using the Person model in AFS, if I select Project Team, the input becomes disabled and then reenabled without the placeholder. Select a different card and then back to Project Team and the widget loads fine.

**Review by chiatt — APPROVED** [review]

**Review by chiatt — CHANGES_REQUESTED** [review]
Very strange, but I'm unable to enter a string with a space in it in the relationship creator UI. 

To reproduce, create a new resource in the resource-instance widget.

image

Then try to enter a string with a space in it and the cursor disappears from the widget and appears in the upper left corner of the page.

image

**Comment by apeters** [comment]
> concept-select does not retain selection
> 
> http://g.recordit.co/ixpQG9Pxvj.gif

fixed in [613207c](https://github.com/archesproject/arches/pull/10265/commits/613207c3508a9f8febfb0e15e01e50b1160b0b63)

**Comment by apeters** [comment]
> The fonts in the account section of the user profile manager are wrong. The values are small and the labels are big:
> 
> Screenshot 2023-11-17 at 3 22 29 PM

fixed by [14b7f86](https://github.com/archesproject/arches/pull/10265/commits/14b7f86160c4b8cb421c4692365d9d0e70bb7f51)

**Review by chiatt — APPROVED** [review]

**Comment by chrabyrd** [comment]
nit: There is no loading UX for dropdowns. When there are several on a page, no results will display until all dropdowns have finished loading. If possible we should include loading UX

http://g.recordit.co/pMWYxQJHql.gif

**Comment by chrabyrd** [comment]
The search results details component pushes the top bar buttons and time filter/saved searches buttons off the page

Screenshot 2023-11-21 at 3 14 52 PM

**Comment by chrabyrd** [comment]
View search-result-details with any advanced search compresses the search results sidebar.

Screenshot 2023-11-21 at 3 17 03 PM

**Comment by chrabyrd** [comment]
The resource instance dropdown in the resource editor list page does not work. Also nit, the `Find` and `Resources` buttons have different font sizes.

Screenshot 2023-11-21 at 3 19 28 PM

**Comment by chrabyrd** [comment]
Sometimes when loading concept-multiselect components, their options will be listed twice:

http://g.recordit.co/X4awBOjlI5.gif

I am able to consistently reproduce this via creating a new `Instrument` resource in the resource editor.

**Comment by chrabyrd** [comment]
deleting a `resource-instance-select` selection does not update the card tree

http://g.recordit.co/4RMKbKNLOv.gif

**Review by chrabyrd — CHANGES_REQUESTED** [review]
Some of the above points are nits and can be ignored, but the duplication of values in concept-multiselect appear to be a serious issue.

**Review by chiatt — CHANGES_REQUESTED** [review]
In the Upload XY Data workflow, the iiif gallery is pushed down below the bottom of the page.
image

Also, I'm not able to select importers in the XY Loader step.

image

**Comment by apeters** [comment]
> The resource instance dropdown in the resource editor list page does not work. Also nit, the `Find` and `Resources` buttons have different font sizes.
> 
> Screenshot 2023-11-21 at 3 19 28 PM

this doesn't work either in dev/7.5.x  (it's also a chosen dropdown which I didn't touch)

**Comment by apeters** [comment]
> The search results details component pushes the top bar buttons and time filter/saved searches buttons off the page
> 
> Screenshot 2023-11-21 at 3 14 52 PM



> View search-result-details with any advanced search compresses the search results sidebar.
> 
> Screenshot 2023-11-21 at 3 17 03 PM

fixed by [05a418d](https://github.com/archesproject/arches/pull/10265/commits/05a418d3979db0f69ca58958bf40ca5e89b81f2f)

**Comment by apeters** [comment]
> In the Upload XY Data workflow, the iiif gallery is pushed down below the bottom of the page. image
> 
> Also, I'm not able to select importers in the XY Loader step.
> 
> image

fixed by [3388b94](https://github.com/archesproject/arches/pull/10265/commits/3388b94b876837f1e08e8a7839bf181db592021d)

**Comment by apeters** [comment]
> Also, I'm not able to select importers in the XY Loader step.
> 
> image

this is probably because the select2 configs are now outdated.  If we fix this now before we push the accessibility branch then this will break for everyone.  I will have to stage a fix on the AFS repo, to be applied after the accessibility branch gets merged in.

**Comment by apeters** [comment]
> deleting a `resource-instance-select` selection does not update the card tree
> 
> http://g.recordit.co/4RMKbKNLOv.gif

this is an issue in dev/7.5.x

**Comment by apeters** [comment]
> Sometimes when loading concept-multiselect components, their options will be listed twice:
> 
> http://g.recordit.co/X4awBOjlI5.gif
> 
> I am able to consistently reproduce this via creating a new `Instrument` resource in the resource editor.

fixed by [d7b8b7c](https://github.com/archesproject/arches/pull/10265/commits/d7b8b7cb78062121875090bda7a87a8b30bef2ff)

**Comment by chrabyrd** [comment]
nit: some concepts show `&nbsp;` 

Screenshot 2023-11-28 at 1 37 02 PM

**Comment by chrabyrd** [comment]
Overall lgtm -- there's still several AFS-level issues but since they've been mentioned already and will be solved in their own ticket I won't repeat them here. 

I appreciate the level of complexity here and thanks for working with us to stamp out all the bugs we can find before merge 👍

**Review by chrabyrd — APPROVED** [review]

**Comment by apeters** [comment]
> nit: some concepts show `&nbsp;`
> 
> Screenshot 2023-11-28 at 1 37 02 PM

fixed in [1d23fb6](https://github.com/archesproject/arches/pull/10265/commits/1d23fb6bc281cd726d7e6b6381cdef29a8da49f9)

**Comment by chrabyrd** [comment]
nit: RDM dropdown does not have placeholder text

Screenshot 2023-11-30 at 2 44 42 PM

**Comment by chrabyrd** [comment]
nit: clearing a value from dropdowns opens the dropdown

http://g.recordit.co/1Z2TqqrkBA.gif

**Comment by chrabyrd** [comment]
nit: concept-multiselect values are not properly aligned

Screenshot 2023-11-30 at 2 52 36 PM

**Comment by chrabyrd** [comment]
All of the above are nits, I haven't found anything that would block merging in these changes 👍

**Comment by apeters** [comment]
> nit: RDM dropdown does not have placeholder text
> 
> Screenshot 2023-11-30 at 2 44 42 PM

fixed in [99122d5](https://github.com/archesproject/arches/pull/10265/commits/99122d54aa3ff1c7becc714a23ccf9437ca720c3)

**Comment by apeters** [comment]
> nit: the larger text in the text/rich text widgets can push the copy/edit/delete buttons out of alignment.
> 
> Additional question -- I noticed that throughout the system _some_ text is larger but not all text. Take the card in the image below for instance. Is this intentional? What's the rule of thumb here?
> 
> Screenshot 2023-11-30 at 2 35 21 PM

@chrabyrd I'm not seeing this on my site.  Can you tell what style it's using?

**Comment by apeters** [comment]
@chrabyrd if there is really no more to QA and it's only small styling issues, then I'd say we shouldn't block testing and merge this into core.

**Comment by chrabyrd** [comment]
@apeters agreed, already approved the changes a few iterations ago

**Pull request merged**
